packages/desktop-client/src/components/util/GenericInput.jsx

```diff
@@ -209,15 +245,21 @@ export function GenericInput({
           />
         );
       } else {
-        content = (
-          <Input
-            inputRef={inputRef}
-            defaultValue={value || ''}
-            placeholder="nothing"
-            onEnter={e => onChange(e.target.value)}
-            onBlur={e => onChange(e.target.value)}
-          />
-        );
+        switch (type) {
+          case 'number':
+            content = getNumberInputByFormatType(numberFormatType);
+            break;
+          default:
```

Comment:
Is this really needed?  What if the default `numberFormatType` was undefined and then here you just call the function?  You might have to find places that call this code and ensure they pass in `currency` as the number format, where it makes sense.  But that seems better anyway to make that more explicit.

Reply:
I added a commit which I think matches what you mean? Let me know

Reply:
(I deliberately omitted the 'filters' page because that page is sign-agnostic)